File: 프로그래머스/unrated/181895. 배열 만들기 3/배열 만들기 3.cs
using System;

public class Solution {
    public int[] solution(int[] arr, int[,] intervals) 
    {
        int size1 = intervals[0, 1] - intervals[0, 0] + 1;
        int size2 = intervals[1, 1] - intervals[1, 0] + 1;
        
        int[] answer = new int[size1 + size2];
        Array.Copy(arr, intervals[0, 0], answer, 0, size1);
        Array.Copy(arr, intervals[1, 0], answer, size1, size2);
        
        return answer;
    }
}

File: 프로그래머스/0/181895. 배열 만들기 3/배열 만들기 3.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr, vector<vector<int>> intervals) {
    vector<int> answer;
    
    for (int i = 0; i < intervals.size(); ++i)
    {
        auto start = arr.begin() + intervals[i][0];        
        auto end = arr.begin() + intervals[i][1] + 1;        
        
        answer.insert(answer.end(), start, end);
    }
    
    return answer;
}